cms/tests/test_page.py

```diff
@@ -743,204 +733,275 @@ def test_plugin_loading_queries(self):
         # Refs: 6346
-        if getattr(settings, 'MIDDLEWARE', None):
+        if getattr(settings, "MIDDLEWARE", None):
             override = {
-                'MIDDLEWARE': settings.MIDDLEWARE + [
-                    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+                "MIDDLEWARE": settings.MIDDLEWARE
+                + [
+                    "django.middleware.clickjacking.XFrameOptionsMiddleware",
                 ]
             }
         else:
             override = {
-                'MIDDLEWARE_CLASSES': settings.MIDDLEWARE_CLASSES + [
-                    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+                "MIDDLEWARE_CLASSES": settings.MIDDLEWARE_CLASSES
+                + [
+                    "django.middleware.clickjacking.XFrameOptionsMiddleware",
                 ]
             }
 
-        override['CMS_PAGE_CACHE'] = True
+        override["CMS_PAGE_CACHE"] = True
 
         with self.settings(**override):
             page = create_page(
-                'test page 1',
-                'nav_playground.html',
-                'en',
+                "test page 1",
+                "nav_playground.html",
+                "en",
                 xframe_options=constants.X_FRAME_OPTIONS_ALLOW,
             )
 
             # Normal response from render_page
-            resp = self.client.get(page.get_absolute_url('en'))
-            self.assertEqual(resp.get('X-Frame-Options'), None)
+            resp = self.client.get(page.get_absolute_url("en"))
+            self.assertEqual(resp.get("X-Frame-Options"), None)
 
             # Response from page cache
-            resp = self.client.get(page.get_absolute_url('en'))
-            self.assertEqual(resp.get('X-Frame-Options'), None)
+            resp = self.client.get(page.get_absolute_url("en"))
+            self.assertEqual(resp.get("X-Frame-Options"), None)
 
+    def test_page_cache_basic(self):
+        """
+        Test that the page cache works in its basic form
+        """
+        page = self.create_homepage("page", "nav_playground.html", "en")
+        page_url = page.get_absolute_url()
+        request = self.get_request(page_url)
 
-class PageTreeTests(CMSTestCase):
+        # First request - should set cache
+        response = self.client.get(page_url)
+        self.assertEqual(response.status_code, 200)
+        cached_data = get_page_cache(request)
+        self.assertIsNotNone(cached_data)
 
+        # Verify cache contents
+        cached_content, cached_headers, expires = cached_data
+        self.assertEqual(cached_content, response.content)
+
+    def test_page_cache_language_handling(self):
+        """
+        Test that the page cache properly handles different languages
+        """
+        page = self.create_homepage("page", "nav_playground.html", "en")
+        create_page_content("fr", "french home", page)
+
+        # Get the page in English
+        en_url = page.get_absolute_url(language="en")
```

Comment:
Are `en_url` and `fr_url` identical? This test should give different cached values even for identical urls (say, root) but different languages.

Reply:
The base URL is identical, but en_url has /en/ at the end and fr_url has /fr/. Is that what we expect?